### PR TITLE
Refactor/improve fingerprinting

### PR DESCRIPTION
**Pull request by bruno-fs**
Small improvements for fingerprinting algorithm

### CODE REVIEW

File: quipucords/fingerprinter/tests_fingerprint_task.py [hunk 1]
```diff
@@ -995,6 +995,31 @@ def test_merge_fingerprint(self):
...
+            all(fp_meta["has_sudo"] for fp_meta in metadata_dict.values()),
+            [
+                fp_name
+                for fp_name, fp_meta in metadata_dict.items()
+                if not fp_meta["has_sudo"]
+            ],
+        )
+        return True
+
+    def test_assert_all_fingerprints_have_sudo(self):
```
Comment (gtanzillo) [inline quipucords/fingerprinter/tests_fingerprint_task.py:1015]: Is this just a test to make sure the test for fingerprints having sudo is working? Sort of a test of the test?

Comment (bruno-fs) [reply quipucords/fingerprinter/tests_fingerprint_task.py:1015]: yep :nerd_face: 

kinda like a who watches the watchmen situation XD

**Review by gtanzillo — APPROVED** [review]
LGTM 👍

**Review by nicolearagao — APPROVED** [review]
LGTM

**Pull request merged**